core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala

```diff
@@ -96,22 +120,13 @@ class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext with L
     taskDescriptions = taskScheduler.resourceOffers(multiCoreWorkerOffers).flatten
     assert(1 === taskDescriptions.length)
     assert("executor0" === taskDescriptions(0).executorId)
+    assert(!failedTaskSet)
   }
 
   test("Scheduler does not crash when tasks are not serializable") {
-    sc = new SparkContext("local", "TaskSchedulerImplSuite")
```

Comment:
mentioned this below as well, but just to be clear -- I was mistaken, that bug doesn't effect the case where the tasks aren't serializable.  That still correctly fails with an error about serialization.  The error I was encountering is in a different case ("multiple CPUs per task", since there you never add the executors, just the hosts), and still needs a workaround for now, which I've added.